typescript-react/example/api/model/domain.ts

```diff
@@ -43,14 +55,165 @@ const packages: IPackageVM[] = [
   },
 ];
 
+const users: IUserVM[] = [
+  {
+    ID: uuid() as UUID,
+    firstName: 'Peregrin' as TextStr,
+    lastName: 'Took' as TextStr,
+    email: 'pippin@test.com' as TextStr,
+    address:  {
+      street: 'Street 1',
+      zipCode: '20024',
+      region: 'Shire',
+      city: 'Hobbit',
+      country: 'Shire',
+    },
+    gender: Gender.Male,
+    status: UserStatus.Active,
+  },
+  {
+    ID: uuid() as UUID,
+    firstName: 'Meriadoc' as TextStr,
+    lastName: 'Brandybuck' as TextStr,
+    address:  {
+      street: 'Test 213',
+      zipCode: '21000',
+      region: 'Shire',
+      city: 'Hobbiton',
+      country: 'Shire',
+    },
+    email: 'merry@test.com' as TextStr,
+    gender: Gender.Male,
+    status: UserStatus.Active,
+  },
+];
+
 const gt = (a?: string | number, b?: string | number) => a != null && b != null && new BigNumber(a).isGreaterThan(new BigNumber(b));
 const lt = (a?: string | number, b?: string | number) => a != null && b != null && new BigNumber(a).isLessThan(new BigNumber(b));
 const empty = (a?: any) => a == null || (typeof a === 'string' && a.trim().length === 0) || ((a as any[]).length === 0);
+const includes = (a: string, b: string) => a.toLowerCase().includes(b.toLowerCase());
 
 type SubmitErrors<T> = {
   [K in keyof T]?: string | SubmitErrors<T[K]>;
 }
 
+const getFilteredUsers = (filter: ISearchUsersFilter): IUserVM[] =>
+  users.filter((u: IUserVM) => {
+    if (!includes(u.firstName, filter.firstName || '')) {
+      return false;
+    }
+
+    if (!includes(u.lastName, filter.lastName || '')) {
+      return false;
+    }
+
+    if (!includes(u.email, filter.email || '')) {
+      return false;
+    }
+
+    if (filter.status != null && filter.status.length > 0 && !filter.status.includes(u.status)) {
+      return false;
+    }
+
+    return true;
+  });
+
+export const searchUsers = (command: ISearchUsers): ISearchUsers => ({
+  ...command,
+  users: getFilteredUsers(command.filter),
+});
+
+const validateUserSubmit = (command: ICreateUser | IEditUser): void => {
+  const errors: SubmitErrors<ICreateUser | IEditUser> = {};
+  appendErrors(errors, 'address', validateAddress(command.address || {}));
+
+  if (empty(command.firstName)) {
+    errors.firstName = 'Required';
+  }
+
+  if (empty(command.lastName)) {
+    errors.lastName = 'Required';
+  }
+
+  if (empty(command.email)) {
+    errors.email = 'Required';
+  }
+
+  if (empty(command.gender)) {
+    errors.gender = 'Required';
+  }
+
+  if (Object.keys(errors).length > 0) {
+    throw errors;
+  }
+}
+
+export const createUser = (command: ICreateUser): ICreateUser => {
+  validateUserSubmit(command);
+  // reusing FE model, so read-only, we're hacking around that
+  (command as any).ID = uuid() as UUID;
```

Comment:
Ok, I was lazy to do this myself, but maaaaybe:
```
type Writeable<T> = { -readonly [P in keyof T]: T[P] };

export const createUser = (command: Writeable<ICreateUser>): ICreateUser => {
  validateUserSubmit(command);
  command.ID = uuid() as UUID;
  ...

```

Reply:
Uhhhhh, pretty sure this _still_ won't work, since `api/index.ts` should complain now, I think? Did you check? I think you need

```
export const registerSubmitHandler = <T>(app: express.Application, name: string, handler: (command: Writeable<T>) => Writeable<T>) => {
```

Reply:
Or to use `Writeable` on the usage of the function side